### PR TITLE
Python 3 support, move from fabric to invoke.  Include a manifest for…

### PR DESCRIPTION
**Pull request by jawnb**
… ship_it so you can ship_it ship_it while you ship_it

### CODE REVIEW

**Comment by robdennis** [comment]
I super wanted to move to invoke, thanks thanks thanks

**Comment by jawnb** [comment]
This also makes an executable called ship_it that you can run to make rpms instead of having to do the from ship_it import fpm thing

File: tests/test_manifest.py [hunk 1]
```diff
@@ -4,7 +4,10 @@
 import mock
 import os
 import pytest
-from StringIO import StringIO
+try:
+    from StringIO import StringIO
+except ImportError:
+    from io import StringIO
```
Comment (robdennis) [inline tests/test_manifest.py]: ``six`` has something for this, but there's not enough 2/3 weirdness to _really_ need to bring it in

File: manifest.yaml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+version: 0.6.0
```
Comment (robdennis) [inline manifest.yaml]: Although manifests are clearly used for the projects we build RPMs with, I'm not sure what _this_ manifest is up to? Is this more of an example?

Comment (jawnb) [inline manifest.yaml]: Yeah, I used it to verify ship_it works, can omit if you'd like

**Comment by robdennis** [comment]
@jawnb the travis run of tox seemed to only actually execute py27, looks like it's because the travis configured like that:
https://github.com/robdennis/ship_it/blob/master/.travis.yml#L3-L7

can you edit/commit the travis.yml? I'm hoping that will retrigger travis with both environments

File: manifest.yaml (hunk 2 above)
Comment (robdennis) [inline manifest.yaml]: I think it mostly highlights an annoying thing I don't like: the need to update version numbers in two places. Not going to fix that for now, and thinking it through, having a manifest in the repo is useful for documentation, so we'll keep it

File: .travis.yml [hunk 3]
```diff
@@ -1,5 +1,7 @@
 language: python
-python: 2.7
+python:
+  - 2.7
+  - 3.5
```
Comment (robdennis) [inline .travis.yml]: I believe this has the effect of running:
- python 2.7 tox runs everything the tox.ini configures (py27 and py35)
- python 3.5 tox runs everything the tox.ini configures (py27 and py35)

fortunately we can see that in action to confirm

Comment (robdennis) [inline .travis.yml]: confirmed, I think we probably want to pick a single python version to run tox, and control which python version is used for the tests using the tox_env environment variable

Comment (jawnb) [inline .travis.yml]: should be fixed now.

File: manifest.yaml (hunk 2 above)
Comment (jawnb) [inline manifest.yaml]: it would be neat to pull in the version from setup.py and use that.

**Comment by jawnb** [comment]
thank you sir, any chance of updating pypi? :-)

**Comment by robdennis** [comment]
https://pypi.python.org/pypi/ship_it/0.6.0

(tried replying via email earlier, but didn't seem to make it to this thread)

**Comment by jawnb** [comment]
yay thank you

**Comment by robdennis** [comment]
no THANK *YOU*

**Comment by robdennis** [comment]
https://pypi.python.org/pypi/ship_it/0.6.0

**Pull request merged**